### ISSUE TITLE
Java processes are not closed when closing Minecraft & GDLauncher

### ISSUE THREAD

**Issue by miccarrer** [killpowa/Twickt-Launcher#1586]
**Describe the bug**
Java processes are not closed when closing Minecraft & GDLauncher, on Artix Linux

**To Reproduce**
Run GDLauncher v1.1.30 thanks to aur package (tried at first) or AppImage.
Install Java thanks to automatic option in GDLauncher (maybe not relevant).
Download and ATM8 modpack.
Use slider to increase Java memory usage to 16Go for the modpack (maybe not relevant)
Start the modpack, enter world.
Quit world.

=> Java processes are still running (16Go memory usage)

**Expected behavior**
I expect Java processes to stop when I close Minecraft and GDLauncher.

**Screenshots**
Process list when everything is closed:
![image](https://github.com/gorilla-devs/GDLauncher/assets/4253471/523789fe-e017-47d6-b5b6-ad755416e2b6)

GDLauncher error during startup (maybe not relevant):
![image](https://github.com/gorilla-devs/GDLauncher/assets/4253471/b16dd4da-eed2-426e-b35d-b52d4e6cb59b)

**Operating System:**
- OS: Artix Linux
- Launcher Version: 1.1.30
- Java Version: 17
- Minecraft Version: 1.19.2
- Mod Loader: Forge
- Mod Loader version: 43.2.14

**Logs:**
[latest.log](https://github.com/gorilla-devs/GDLauncher/files/12430308/latest.log)

**Comment by TomPillet** [comment]
Hi

I started to get the same problem since a few weeks ago. This is quite annoying, forcing to go on system manager and kill java process manually.

I'm on Fedora 38, same versions as @miccarrer for java, the launcher and the instance (except I play on Forge 43.2.13) 

Edit : apparently, this bug doesn't appear on Vanilla, and might be related to forge or specific mod that crash during game shutdown. Note that kill instance via GDLauncher, instead of using "Quit game" button in game, seems don't create the bug.

**Comment by miccarrer** [comment]
Good tip Tom! It kills java processes indeed.
But then I get an oopsie woopsie error (Error: Signal Exits, code: 143)

**Comment by Eskaan** [comment]
So, firstly: The permission error on startup is typical for a protected file system (like an AppImage), and shouldn't really affect the launcher. Secondly, it is not intended for the game to close when closing the launcher and I think this mimics the Microsoft launcher's behavior.

What should happen is that closing the game kills all child processes of it.
I read that you quit the world, so I assume you mean you quit the game as a whole.

I would recommend you try the same with a fresh forge instance (on the same version) to check if this is a launcher issue or not. This doesn't really sound like it as Java (Minecraft) should handle killing childs on it's own.

Thanks for a very good issue report btw, I rarely get those on this issue tracker :+1:

**Comment by TGurl** [comment]
I don't think this is a launcher issue. I don't experience these problems with all modpacks. I did a small test and as far as I can detect it only occurs with modpacks utilizing the Rubidium and/or Oculus  mods. I am not sure this is a mod issue either as this could very well be a kernel issue as well, just wanted to give some more insight.

I did a small test (attached below).

[log.txt](https://github.com/gorilla-devs/GDLauncher/files/12830167/log.txt)
